### PR TITLE
Add basic CI

### LINKED ISSUE #1: Add CI infrastructure
Basic CI to make sure the project builds

### PR DESCRIPTION
**Pull request by muzzammilshahid**
closes #1

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+name: main
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
```
Comment (Mahad-10) [inline .github/workflows/main.yml]:
```suggestion
  push:
    branches:
      - main
  pull_request:
    branches:
      - main
```
you can simplify this, here's an example https://github.com/xconnio/wampproto.py/blob/main/.github/workflows/main.yaml

**Review by Mahad-10 — CHANGES_REQUESTED** [review]

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+name: main
+
+on: [push, pull_request]
+
+jobs:
+  check:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
```
Comment (Mahad-10) [inline .github/workflows/main.yml]:
```suggestion
      - uses: actions/checkout@v4
```

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+      - name: Set up JDK 17
+        uses: actions/setup-java@v3
```
Comment (Mahad-10) [inline .github/workflows/main.yml]:
```suggestion
        uses: actions/setup-java@v4
```

**Review by Mahad-10 — APPROVED** [review]
LGTM

**Review by om26er — APPROVED** [review]

**Pull request merged**